cfgov/external_links/views.py

```diff
@@ -0,0 +1,76 @@
+from django.shortcuts import render
+from django.views.generic import View
+
+from external_links.forms import ExternalLinksForm
+
+from ask_cfpb.models.pages import AnswerPage
+from v1.models.base import CFGOVPage
+from v1.models.blog_page import BlogPage, LegacyBlogPage
+from v1.models.browse_filterable_page import BrowseFilterablePage
+from v1.models.browse_page import BrowsePage
+from v1.models.home_page import HomePage
+from v1.models.landing_page import LandingPage
+from v1.models.learn_page import (
+    AbstractFilterPage, DocumentDetailPage, EventPage, LearnPage
+)
+from v1.models.snippets import Contact, Resource, ReusableText
+from v1.models.sublanding_filterable_page import SublandingFilterablePage
+from v1.models.sublanding_page import SublandingPage
+
+
+class SearchView(View):
+    template_name = 'search.html'
+    # These are all v1 models that directly implement a visible part
+    # of the Wagtail page, specifically the content, sidefoot, or header
+    v1_page_models = [
+        CFGOVPage, HomePage, AbstractFilterPage, BlogPage,
+        BrowsePage, BrowseFilterablePage,
+        DocumentDetailPage, EventPage,
+        LegacyBlogPage, LearnPage, LandingPage,
+        SublandingPage, SublandingFilterablePage,
+    ]
+
+    def get(self, request):
+        return render(request, self.template_name, {
+            'form': ExternalLinksForm()
+        })
+
+    def post(self, request):
+        form = ExternalLinksForm(request.POST)
+        if form.errors:
+            return render(request, self.template_name, {
+                'form': form
+            })
+
+        url = form.data['url']
+        pages = []
+
+        for cls in self.v1_page_models:
+            pages += list(cls.objects.search(url))
```

Comment:
I don't think so, because the search indexes aren't redundant. In your example, `CFGOVPage` will return results based on the `sidefoot` for all pages that inherit from it, `AbstractFilterPage` will return results based on the `header` for all pages that inherit from it, and `EventPage` will return results based on the search fields specific to it.

Reply:
Oh - I see what you're saying now.  Sorry.   Yeah, I guess if a page had matching results in all of those different sections of the page, then it could end up in duplicates, since it's not like there is a field in the search results for where it shows up in the page.  Let me look into that.

Reply:
I can confirm there will be "duplicates" in the results if the same URL appears in multiple page sections, e.g. both the content and the sidefoot.  It seems like a bit of an edge case - I could do some hacks to remove the duplicates from the list but the additional processing time for that seems not worth it.  @ielerol how bad is it if the list occasionally has repeated pages?  I actually don't consider them true duplicates since it represents the instances that URL appears on the page, but I do get that it's confusing since they can't see which part of the page it is.

Reply:
Hm, I think we actually want to display the duplicates in the list so editors know to check several sections, otherwise someone might find the link in the content and not think to check the sidefoot also, or wherever else. But it's something we might need some explanatory text for. Is it possible to add helper text at the bottom of the page list that says something about what it means for a page to show up multiple times?

Reply:
@ielerol Definitely can add some help text!